prov/gni/src/gnix_mr.c

```diff
@@ -486,6 +486,9 @@ static int __udreg_init(struct gnix_fid_domain *domain)
 				"Could not access udreg application cache, urc=%d",
 				urc);
 	}
+
+	domain->mr_is_init = 1;
+
 	return 0;
```

Comment:
Sorry I did not catch this before...  to be consistent, this should return FI_SUCCESS.